### PR TITLE
pypi module files

### PR DESCRIPTION
**Pull request by gal9**

### CODE REVIEW

**Comment by jO-Osko** [comment]
This is nice PR with a working pypi distribution, good work 👍 

There are some minor stylistic comments, but the functionality is good.

- Use `pep8`, linter, or something similar in your editor. This informs you about formatting issues. They are not important for code correctness but can hinder readability and make maintenance a nightmare.

- Always provide a description of what a pull request adds. In your case commit message is informative enough, but for a bigger one, it is important to have a description of what has changed. 

Requested changes are added as comments to specific files.

The most important question to be answered is the one regarding the project structure.

File: Example.ipynb [hunk 1]
```diff
@@ -112,7 +109,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
```
Comment (jO-Osko) [inline Example.ipynb]: Before commiting file, either clear inputs or rerun it, so that either cell numbers are sequential or there is no output at all.

File: MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1 @@
+include README.md LICENSE
```
Comment (jO-Osko) [inline MANIFEST.in]: Add newline

File: README.md [hunk 3]
```diff
@@ -15,12 +15,18 @@ For further details see `requirements.txt`.
 * For example:
```
Comment (jO-Osko) [inline README.md:11]: As in the `setup.py`, it would be nice to correct `README.md` as it also lists full requirements. You should just list the general ones.

File: README.md [hunk 4]
```diff
@@ -15,12 +15,18 @@ For further details see `requirements.txt`.
...
     * Pandas
 
+## Installation
```
Comment (jO-Osko) [inline README.md:16]: The `pandas` library is optional and only used in examples.  A nice way to include this into `setup.py` is to use `extras_require` argument for `setup` function. See https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html for example. 

Users can then install it using `pip install fastener[examples]`.  It would be nice to showcase and explain this in Installation part.

File: README.md [hunk 5]
```diff
@@ -82,23 +85,26 @@ def eval_fun(model: Any, genes: "Genes", shuffle_indices: Optional[List[int]] =
     return res
 ```
 
-3) Configure the FASTENER
+3. Configure the FASTENER
 
 By default fastener runs for 1000 iterations. The number of iterations can be adjusted with `number_of_rounds` parameter in the `fastener.Config()`.
```
Comment (jO-Osko) [inline README.md]: Remove `()` from `fastener.Config()`

File: README.md [hunk 6]
```diff
@@ -82,23 +85,26 @@ def eval_fun(model: Any, genes: "Genes", shuffle_indices: Optional[List[int]] =
...
 
 ```python
```
Comment (jO-Osko) [inline README.md:90]: Adhere to pep8 standards in examples: 
- Use whitespace around operators: `a = 3`
- Comments should start with "# "
- There is no whitespace around keyword arguments, except when types are specified

File: README.md [hunk 7]
```diff
@@ -139,12 +145,31 @@ for item in best:
...
+```
+* Intersection mating: If both of the parents have the feature the descendent will have it too.
+```python
+mating_strategy=IntersectionMating()
+```
+* Intersection mating with information gain: If both of the parents have the feature the descendent will have it too. Additionally, some features from either one of the parents, that have the highest information gain are added.  
+```python
+mating_strategy=IntersectionMatingWithInformationGain()
+```
+* Intersection mating with weighted random information gain: If both of the parents have the feature the descendent will have it too. Additionally, some features from either one of the parents, will be added. The probbability of selecting a features is proportionate to this feature's information gain.
```
Comment (jO-Osko) [inline README.md]: This line might be misleading (it is similarly misleading in the docstring). The probability of a feature being selected is proportional to scaling function applied to it's calculated information gain. One probability has too many b-s.

File: setup.py [hunk 8]
```diff
@@ -1,10 +1,32 @@
...
-    description='desc',
+    version='1.0.3',
     author='Filip Koprivec, Klemen Kenda, Gal Petkovsek',
+    author_email='',
+    description='Feature selection enabled by entropy',
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/JozefStefanInstitute/FASTENER.git",
+    packages=["fastener"],
+    #py_modules=["fastener", "item", "random_utils"],
```
Comment (jO-Osko) [inline setup.py]: Remove comment

File: setup.py [hunk 9]
```diff
@@ -1,10 +1,32 @@
...
+    install_requires=["joblib==0.14.1",
```
Comment (jO-Osko) [inline setup.py]: Install requires is used only for top-level requirements. In your case, you only need `mypy` as `mypy-extensions` and `typed-ast` are required by mypy. Similarly, remove other unneeded libraries. For additional explanation see: https://packaging.python.org/discussions/install-requires-vs-requirements/.

File: setup.py [hunk 10]
```diff
@@ -1,10 +1,32 @@
...
+                    "mypy==0.761",
+                    "mypy-extensions==0.4.3",
+                    "numpy==1.18.1",
+                    "scikit-learn>=0.22.2",
+                    "scipy>=1.4.1",
+                    "typed-ast>=1.4.1",
+                    "typing-extensions>=3.7.4.1"],
+    classifiers=[
```
Comment (jO-Osko) [inline setup.py:23]: Add more classifiers

File: setup.py [hunk 11]
```diff
@@ -1,10 +1,32 @@
...
     author='Filip Koprivec, Klemen Kenda, Gal Petkovsek',
+    author_email='',
+    description='Feature selection enabled by entropy',
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/JozefStefanInstitute/FASTENER.git",
+    packages=["fastener"],
+    #py_modules=["fastener", "item", "random_utils"],
+    install_requires=["joblib==0.14.1",
+                    "mypy==0.761",
```
Comment (jO-Osko) [inline setup.py]: Fix hanging indent.
Be consistent when using single or double quotes in the same file.

File: setup.py [hunk 12]
```diff
@@ -1,10 +1,32 @@
...
+                    "scipy>=1.4.1",
+                    "typed-ast>=1.4.1",
+                    "typing-extensions>=3.7.4.1"],
+    classifiers=[
+        "Programming Language :: Python :: 3.6",
+        "License :: OSI Approved :: MIT License",
+    ],
     license='MIT',
+    python_requires='>=3.6',
+    package_dir = {'': 'src'},
```
Comment (jO-Osko) [inline setup.py]: Remove whitespace and add a newline at the end.

File: src/fastener/fastener.py [hunk 13]
```diff
@@ -9,14 +9,14 @@
...
-from src.random_utils import shuffle
+import fastener.random_utils as random_utils
+from fastener.random_utils import shuffle
 from typing import Dict, List, Callable, Any, Tuple, Optional, \
     Counter as CounterType, Set
 import numpy as np
 from collections import Counter
 
-from src.item import Item, Result, Population, flatten_population, FitnessFunction, \
+from fastener.item import Item, Result, Population, flatten_population, FitnessFunction, \
```
Comment (jO-Osko) [inline src/fastener/fastener.py]: Some of the imports are no longer needed, consider cleaning them up.

File: src/fastener/item.py [hunk 14]
```diff
@@ -9,7 +9,7 @@
 import sklearn.feature_selection
 import numpy as np
 
-from src import random_utils
+import fastener.random_utils as random_utils
```
Comment (jO-Osko) [inline src/fastener/item.py]: Apart from keeping dummy data in a separate folder, is there a reason to move fastener into `src` directory? 

It results in the example file only working with the library installed, but not with the library from the cloned repository.

**Review by jO-Osko — APPROVED** [review]
